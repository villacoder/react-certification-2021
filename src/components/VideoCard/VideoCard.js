import React, { useContext } from 'react';
import Grid from '@material-ui/core/Grid';
import { AppContext } from '../../providers/App/AppProvider';

const VideoCard = ({ changeSelection, videoList, selectedVideoId }) => {
  const { app } = useContext(AppContext);
  const { darkMode } = app;
  const changeSelectedVideo = (data) => {
    changeSelection(data);
  };

  return (
    <>
      {videoList.length > 0 &&
        videoList.map((data) => {
          if (data?.id !== selectedVideoId) {
            return (
              <Grid
                item
                xs={8}
                sm={10}
                md={4}
                lg={3}
                xl={3}
                key={data?.id}
                onClick={() => {
                  changeSelectedVideo(data);
                }}
                style={{ cursor: 'pointer' }}
              >
                <div className="backgroundColor" data-theme={darkMode ? 'dark' : 'light'}>
                  <div className="card" data-theme={darkMode ? 'dark' : 'light'}>
                    <div className="card-cover-image">
                      <img src={data?.thumbnails.medium.url} alt="thumbnail" />
                    </div>
                    <div className="card-box" data-theme={darkMode ? 'dark' : 'light'}>
                      <div className="card-title" data-theme={darkMode ? 'dark' : 'light'}>
                        <span>{data?.title}</span>
                      </div>
                      <div className="card-subtitle" data-theme={darkMode ? 'dark' : 'light'}>{data?.channel.title}</div>
                      <div className="card-content" data-theme={darkMode ? 'dark' : 'light'}>
                        <p>{data?.description}</p>
                      </div>
                    </div>
                  </div>
                </div>
              </Grid>
            );
          }
        })}
    </>
  );
};

export default VideoCard;
