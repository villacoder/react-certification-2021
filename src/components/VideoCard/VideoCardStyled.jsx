import styled from 'styled-components';

const Video = styled.div`
  width: 345px;
  height: 345px;
  padding: 0.5rem;
`;

const Card = styled.div`
  height: 100%;
  border-radius: 5px;
  overflow: hidden;
  border: 1px solid #d3d3d3;
  color: #808080;
  box-shadow: 0 1.3px 1px rgba(0, 0, 0, 0.2);
`;

const CardImageContainer = styled.div`
  height: 140px;
  overflow: hidden;
`;

const CardImage = styled.img`
  width: 100%;
  object-position: 0 -50px;
`;

const CardText = styled.div`
  padding: 1rem;

  h3 {
    line-height: 1.7rem;
    margin-bottom: 1rem;
  }
`;

export { Video, Card, CardImageContainer, CardImage, CardText };
