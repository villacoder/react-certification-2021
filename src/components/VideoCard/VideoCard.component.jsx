import React from 'react';
import { Link } from 'react-router-dom';
import { Video, Card, CardImageContainer, CardImage, CardText } from './VideoCardStyled';

const VideoCard = (video) => {
  return (
    <Link className="no-decoration-text" to={`./videoview/${video.id.videoId}`}>
      <Video>
        <Card>
          <CardImageContainer>
            <CardImage
              src={video.snippet.thumbnails.high.url}
              alt={video.snippet.title}
            />
          </CardImageContainer>
          <CardText>
            <h3>{video.snippet.title}</h3>
            <p>{video.snippet.description}</p>
          </CardText>
        </Card>
      </Video>
    </Link>
  );
};

export default VideoCard;
