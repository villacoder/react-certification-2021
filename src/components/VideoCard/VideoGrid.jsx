import React, { useEffect, useState, useContext } from 'react';
import YouTube from 'simple-youtube-api';
import Grid from '@material-ui/core/Grid';
import VideoCard from './VideoCard';
import { AppContext } from '../../providers/App/AppProvider';
import { types } from '../../providers/App/types';

const youtube = new YouTube(process.env.REACT_APP_API_KEY_YOUTUBE);

export const VideoGrid = () => {
  const { app, dispatch } = useContext(AppContext);
  const { passToChild, selectedVideo } = app;

  const [videoList, setVideoList] = useState([]);
  const [error, setError] = useState(false);

  useEffect(() => {
    const searchStringToUse =
      passToChild || localStorage.getItem('lastSearch') || 'Wizeline';
    const callApi = async () => {
      const result = await youtube.searchVideos(searchStringToUse, 20);

      if (result.length === 0) {
        setError(true);
      } else {
        setError(false);
      }
      setVideoList(result);
    };
    callApi();
    localStorage.setItem('lastSearch', passToChild);
  }, [passToChild, selectedVideo]);

  const selectedVideoCallback = (videoDetail) => {
    dispatch({
      type: types.setPlayVideo,
      payload: true,
    });
    dispatch({
      type: types.setSearchButton,
      payload: false,
    });
    dispatch({
      type: types.setSelectedVideo,
      payload: videoDetail,
    });
  };

  return (
    <Grid sx={{ flexGrow: 1 }} container>
      {!error && (
        <Grid container justifyContent="center">
          <Grid container key={selectedVideo?.id} item justifyContent="center">
            <VideoCard
              changeSelection={selectedVideoCallback}
              videoList={videoList}
              selectedVideoId={selectedVideo?.id}
            />
          </Grid>
        </Grid>
      )}
    </Grid>
  );
};
