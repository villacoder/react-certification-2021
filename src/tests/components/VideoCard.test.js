import React from 'react';
import { render } from '@testing-library/react';
import VideoCard from '../../components/VideoCard';
import { data } from '../../data/youtube-videos-mock';

describe('Testing VideoCard component.', () => {
  const video = data.items[0];

  test('Rendering correctly.', () => {
    const card = render(
      <VideoCard key={video.snippet.thumbnails.default.url} {...video} />
    );

    expect(card).toMatchSnapshot();
  });
});
