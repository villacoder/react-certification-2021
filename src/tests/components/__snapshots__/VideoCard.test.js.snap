// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing VideoCard component. Rendering correctly. 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdnxRM jpBjMb"
      >
        <div
          class="sc-gtsrHT bJomzP"
        >
          <div
            class="sc-dlnjwi lmWHcA"
          >
            <img
              alt="Wizeline"
              class="sc-hKFxyN YJBRr"
              src="https://yt3.ggpht.com/ytc/AAUvwnighSReQlmHl_S_vSfvnWBAG5Cw4A0YxtE0tm5OpQ=s800-c-k-c0xffffffff-no-rj-mo"
            />
          </div>
          <div
            class="sc-eCApnc bqdhCr"
          >
            <h3>
              Wizeline
            </h3>
            <p>
              Wizeline transforms how teams build technology. Its customers accelerate the delivery of innovative products with proven solutions, which combine Wizeline's ...
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdnxRM jpBjMb"
    >
      <div
        class="sc-gtsrHT bJomzP"
      >
        <div
          class="sc-dlnjwi lmWHcA"
        >
          <img
            alt="Wizeline"
            class="sc-hKFxyN YJBRr"
            src="https://yt3.ggpht.com/ytc/AAUvwnighSReQlmHl_S_vSfvnWBAG5Cw4A0YxtE0tm5OpQ=s800-c-k-c0xffffffff-no-rj-mo"
          />
        </div>
        <div
          class="sc-eCApnc bqdhCr"
        >
          <h3>
            Wizeline
          </h3>
          <p>
            Wizeline transforms how teams build technology. Its customers accelerate the delivery of innovative products with proven solutions, which combine Wizeline's ...
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
